### PR TITLE
[Cleanup] Remove internal tools dependency on relative path

### LINKED ISSUE #3062: PRs opened by dependabot for  /hack/internal/tools fail for non-breaking changes
**What would you like to be cleaned**:

PRs opened by dependabot should not be failing, unless for breaking changes in the libraries. In our case they are failing for non-breaking changes. Examples:
- https://github.com/kubernetes-sigs/kueue/pull/3060
- https://github.com/kubernetes-sigs/kueue/pull/3061

Most often they fail on `pull-kueue-verify-main` because the bump is inconsitent. We bump in `/hack/internal/tools`, but not at the root.

A research of potential solutions is welcome. One idea is not to depend on the top-level dir in `/hack/internal/tools` by replace (which is not recognized by dependabot: https://github.com/kubernetes-sigs/kueue/blob/669d73d40e058c237c3ff3f8b820b75b1dc63a08/hack/internal/tools/go.mod#L25).

**Why is this needed**:

The PRs opened by dependabot still require manual chore, more than needed.

### PR DESCRIPTION
**Pull request by mszadkow**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Dependabot suggests updates that are incompatible with internal tools, replaced version.

#### Which issue(s) this PR fixes:

Fixes #3062 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by mbobrovskyi** [comment]
We also need to fix it for kjobctl.

**Comment by mszadkow** [comment]
> We also need to fix it for kjobctl.

I see, thanks

**Review by mbobrovskyi — COMMENTED** [review]
Did you generate docs for kjobctl?

File: cmd/kueuectl-docs/templates/main.html [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 {{ html `<!--
 The file is auto-generated from the Go source code of the component using the
-[generator](https://github.com/kubernetes-sigs/kueue/tree/main/hack/internal/tools/kueuectl-docs).
+[generator](https://github.com/kubernetes-sigs/kueue/tree/cmd/kueuectl-docs).
```
[the remark this answers is not in the document]
Comment (mszadkow) [reply cmd/kueuectl-docs/templates/main.html:10]: maybe you are right, I must have "eaten" that

**Comment by mszadkow** [comment]
> Did you generate docs for kjobctl?

I run both kjobctl generate targets

**Comment by mbobrovskyi** [comment]
> > Did you generate docs for kjobctl?
> 
> I run both kjobctl generate targets

It's strange that it doesn't regenerate docs for kjobctl.

**Comment by mszadkow** [comment]
> > > Did you generate docs for kjobctl?
> > 
> > 
> > I run both kjobctl generate targets
> 
> It's strange that it doesn't regenerate docs for kjobctl.

I found my mistake, they are regenerated rn

**Comment by mbobrovskyi** [comment]
/lgtm

Thanks!

**Comment by mimowo** [comment]
/approve
Awesome, thank you for working on it. I guess the best was to test will be to see the new wave of PRs :)

**Pull request merged**